src/Character.ts

```diff
@@ -8,84 +8,26 @@ import {
 } from './Util/PointUtils'
 
 export class Character {
-  readonly stats: Stats = new Stats()
-  readonly bonuses: Bonuses = new Bonuses()
-  private abilityPoints: number = 30
-  private level: number = 1
-  private isCreating: boolean = true
-  private trainingPoints: number = 0
+  readonly stats: Stats
+  readonly bonuses: Bonuses
+  private abilityPoints = 30
+  private trainingPoints = 0
+  private level = 1
+  private isCreating = true
 
-  //region add stats
-
-  addDexterity(dexterityToAdd: number): void {
-    if (dexterityToAdd > this.abilityPoints) {
-      throw new AbilityPointError('not enough points remaining')
-    }
-    this.abilityPoints -= dexterityToAdd
-    this.stats.dexterity += dexterityToAdd
-  }
-
-  addStrength(strengthToAdd: number): void {
-    if (strengthToAdd > this.abilityPoints) {
-      throw new AbilityPointError('not enough points remaining')
-    }
-    this.abilityPoints -= strengthToAdd
-    this.stats.strength += strengthToAdd
+  constructor() {
+    this.stats = new Stats(this)
+    this.bonuses = new Bonuses()
   }
 
-  addIntelligence(intelligenceToAdd: number): void {
-    if (intelligenceToAdd > this.abilityPoints) {
+  allocateStats(statsToAdd: number): void {
```

Comment:
this is public since the `Stats` class needs to call it. I think, though, that I'm going to pass a callback down into `Stats` instead of the whole character since this is prone to misuse.